File: JavaScript/8js 1/elevator.js
/*
|--------------------------------------------------------------------------
| Closest elevator
|--------------------------------------------------------------------------
|
*/

function elevator(left, right, call) {
   return call == right || Math.abs(call - right) <= Math.abs(call - left) ? 'right' : 'left'
}


function elevator(left, right, call) {
   return (call - left) ** 2 < (call - right) ** 2 ? "left" : "right"
}


function elevator(left, right, call) {
   return Math.abs(call - left) < Math.abs(call - right) ? 'left' : 'right';
}


console.log(elevator(0, 1, 0), 'left');
console.log(elevator(0, 1, 1), 'right');
console.log(elevator(0, 1, 2), 'right');
console.log(elevator(0, 0, 0), 'right');
console.log(elevator(0, 2, 1), 'right');

File: Python/8py 1/elevator.py
"""
|--------------------------------------------------------------------------
| Closest elevator
|--------------------------------------------------------------------------
|
"""

def elevator(left, right, call):
   return 'left' if abs(call-left) < abs(call-right) else 'right'


def elevator(x, y, r):
   return ['left','right'][abs(r-x)>=abs(r-y)]


def elevator(x, y, r):
   return 'rliegfhtt'[abs(r-x)<abs(r-y)::2]


print(elevator(0, 1, 0), "left")
print(elevator(0, 1, 1), "right")
print(elevator(0, 1, 2), "right")
print(elevator(0, 0, 0), "right")
print(elevator(0, 2, 1), "right")